#!/usr/bin/env python
# -*- coding: utf-8 -*-
# 导入包
from dynamic_reconfigure.parameter_generator_catkin import *
PACKAGE = "sentry_chassis_controller"
# 创建参数生成器
gen = ParameterGenerator()

# 左前轮 PID 参数
gen.add("front_left_wheel_p", double_t, 0, "front_left_wheel P", 1.0, 0.0, 10.0)
gen.add("front_left_wheel_i", double_t, 0, "front_left_wheel I", 0.0, 0.0, 1.0)
gen.add("front_left_wheel_d", double_t, 0, "front_left_wheel D", 0.0, 0.0, 1.0)
gen.add("front_left_wheel_i_max", double_t, 0, "front_left_wheel I_max", 2.0, 0.0, 10.0)
gen.add("front_left_wheel_i_min", double_t, 0, "front_left_wheel I_min", -2.0, -10.0, 0.0)
# 右前轮 PID 参数
gen.add("front_right_wheel_p", double_t, 0, "front_right_wheel P", 1.0, 0.0, 10.0)
gen.add("front_right_wheel_i", double_t, 0, "front_right_wheel I", 0.0, 0.0, 1.0)
gen.add("front_right_wheel_d", double_t, 0, "front_right_wheel D", 0.0, 0.0, 1.0)
gen.add("front_right_wheel_i_max", double_t, 0, "front_right_wheel I_max", 2.0, 0.0, 10.0)
gen.add("front_right_wheel_i_min", double_t, 0, "front_right_wheel I_min", -2.0, -10.0, 0.0)
# 左后轮 PID 参数
gen.add("back_left_wheel_p", double_t, 0, "back_left_wheel P", 1.0, 0.0, 10.0)
gen.add("back_left_wheel_i", double_t, 0, "back_left_wheel I", 0.0, 0.0, 1.0)
gen.add("back_left_wheel_d", double_t, 0, "back_left_wheel D", 0.0, 0.0, 1.0)
gen.add("back_left_wheel_i_max", double_t, 0, "back_left_wheel I_max", 2.0, 0.0, 10.0)
gen.add("back_left_wheel_i_min", double_t, 0, "back_left_wheel I_min", -2.0, -10.0, 0.0)
# 右后轮 PID 参数
gen.add("back_right_wheel_p", double_t, 0, "back_right_wheel P", 1.0, 0.0, 10.0)
gen.add("back_right_wheel_i", double_t, 0, "back_right_wheel I", 0.0, 0.0, 1.0)
gen.add("back_right_wheel_d", double_t, 0, "back_right_wheel D", 0.0, 0.0, 1.0)
gen.add("back_right_wheel_i_max", double_t, 0, "back_right_wheel I_max", 2.0, 0.0, 10.0)
gen.add("back_right_wheel_i_min", double_t, 0, "back_right_wheel I_min", -2.0, -10.0, 0.0)
# 左前转向 PID 参数
gen.add("front_left_pivot_p", double_t, 0, "front_left_pivot P", 1.0, 0.0, 10.0)
gen.add("front_left_pivot_i", double_t, 0, "front_left_pivot I", 0.0, 0.0, 1.0)
gen.add("front_left_pivot_d", double_t, 0, "front_left_pivot D", 0.0, 0.0, 1.0)
gen.add("front_left_pivot_i_max", double_t, 0, "front_left_pivot I_max", 2.0, 0.0, 10.0)
gen.add("front_left_pivot_i_min", double_t, 0, "front_left_pivot I_min", -2.0, -10.0, 0.0)   
# 右前转向 PID 参数
gen.add("front_right_pivot_p", double_t, 0, "front_right_pivot P", 1.0, 0.0, 10.0)
gen.add("front_right_pivot_i", double_t, 0, "front_right_pivot I", 0.0, 0.0, 1.0)
gen.add("front_right_pivot_d", double_t, 0, "front_right_pivot D", 0.0, 0.0, 1.0)
gen.add("front_right_pivot_i_max", double_t, 0, "front_right_pivot I_max", 2.0, 0.0, 10.0)
gen.add("front_right_pivot_i_min", double_t, 0, "front_right_pivot I_min", -2.0, -10.0, 0.0)
# 左后转向 PID 参数
gen.add("back_left_pivot_p", double_t, 0, "back_left_pivot P", 1.0, 0.0, 10.0)
gen.add("back_left_pivot_i", double_t, 0, "back_left_pivot I", 0.0, 0.0, 1.0)
gen.add("back_left_pivot_d", double_t, 0, "back_left_pivot D", 0.0, 0.0, 1.0)
gen.add("back_left_pivot_i_max", double_t, 0, "back_left_pivot I_max", 2.0, 0.0, 10.0)
gen.add("back_left_pivot_i_min", double_t, 0, "back_left_pivot I_min", -2.0, -10.0, 0.0)
# 右后转向 PID 参数
gen.add("back_right_pivot_p", double_t, 0, "back_right_pivot P", 1.0, 0.0, 10.0)
gen.add("back_right_pivot_i", double_t, 0, "back_right_pivot I", 0.0, 0.0, 1.0)
gen.add("back_right_pivot_d", double_t, 0, "back_right_pivot D", 0.0, 0.0, 1.0)
gen.add("back_right_pivot_i_max", double_t, 0, "back_right_pivot I_max", 2.0, 0.0, 10.0)
gen.add("back_right_pivot_i_min", double_t, 0, "back_right_pivot I_min", -2.0, -10.0, 0.0)

# 生成配置文件
exit(gen.generate(PACKAGE, "sentry_chassis_controller", "SentryChassisController"))